src/Screencast/main.vala

```diff
@@ -0,0 +1,81 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1 --pkg libportal-gtk4 --pkg gstreamer-1.0
+
+private Gtk.Picture output;
+
+async void start_screencast_session (Xdp.Portal portal, Xdp.Parent parent) {
+    try {
+        Xdp.Session session = yield portal.create_screencast_session (Xdp.OutputType.MONITOR,
+            Xdp.ScreencastFlags.NONE,
+            Xdp.CursorMode.EMBEDDED,
+            Xdp.PersistMode.TRANSIENT,
+            null,
+            null);
+
+        if (session == null) {
+            message ("Permission denied");
+            return;
+        }
+
+        var success = yield session.start (parent, null);
+
+        if (!success) {
+            message ("Could not start session");
+            return;
+        }
+        var pw_remote = session.open_pipewire_remote ();
+        var pipeline = new Gst.Pipeline ("");
+        var source = Gst.ElementFactory.make ("pipewiresrc", "source");
+        var queue = Gst.ElementFactory.make ("queue", "queue"); // add a queue element
+        var paintable_sink = Gst.ElementFactory.make (
+                                                      "gtk4paintablesink",
+                                                      "paintable_sink"
+        );
+        var glsinkbin = Gst.ElementFactory.make ("glsinkbin", "glsinkbin");
+
+        // Set up and Link Pipeline
+        source.set_property ("fd", pw_remote); // pw_remote is the file descriptor obtained from libportal
+
+        // Obtain the node id from the screencast session
+        var streamss = session.get_streams ();
+        var node_id = streamss.get_child_value (0);
```

Comment:
Also, the variant returned by streams is an array of tuples (in the form a(ua{sv}). Which means you would have to do:

```suggestion
        uint node_id = streamss.get_child_value (0).get_child_value (0).get_uint32 ();
```

Where the first get_child_value gets the first tuple in the array and the second one gets the first element of the tuple. Variants are very useful but sometimes their representation with letters is a little intimidating. You can take a look at the [Variant documentation](https://docs.gtk.org/glib/gvariant-format-strings.html#arrays), which contains very useful information to understand them.